### ISSUE TITLE
EMQX hangs on Startup when trying to rejoin the cluster

### ISSUE THREAD

**Issue by jkominiakgps** [emqx/emqx#11218]
### What happened?

A replicant node in our cluster seemingly crashed.  After rebooting the VM and Reimaging it, it is still unable to join the cluster again.  When running cli commands, the node itself reports as "starting" and the cluster status reports the node as "running" however, the node is not part of the cluster.  Looking at the `syslog`, I can see the start command for `emqx` but then nothing else is ever reported there even after 15 minutes.  Prior to this, we've had no issues with the cluster or nodes leaving/joining.
Any insight you could provide into this would be greatly appreciated as we're unable to replace the downed node, or add new nodes to the cluster at this point.  As a side note - we did have a few `rules/flows` configured, but the `data bridge` was disabled.

I see errors like this when viewing the dashboard: `500 NODE_DOWN:bad rpc call 'emqx@XXX', Reason {'EXIT', {badarg, [{ets,select_count, [emqx_activated_alarm, [{'$1',[],[true]}]], [{error_info, #{cause => id, module => erl_stdlib_errors}}]}, {emqx_mgmt_api, '-counting_total_fun/1-fun-0-',2, [{file,"emqx_mgmt_api.erl"}, {line,357}]}, {emqx_mgmt_api, maybe_apply_total_query,2, [{file,"emqx_mgmt_api.erl"}, {line,333}]}, {emqx_mgmt_api,do_select,2, [{file,"emqx_mgmt_api.erl"}, {line,299}]}, {emqx_mgmt_api,do_query,2,[]}]}}`

And in the log file I see
```
[error] State machine '$mria_meta_shard' terminating. Reason: {timeout,{gen_server,call,[mria_lb,core_nodes,30000]}}. Stack: [{gen_server,call,3,[{file,"gen_server.erl"},{line,247}]},{mria_rlog_replica,try_connect,2,[{file,"mria_rlog_replica.erl"},{line,378}]},{mria_rlog_replica,handle_reconnect,1,[{file,"mria_rlog_replica.erl"},{line,341}]},{gen_statem,loop_state_callback,11,[{file,"gen_statem.erl"},{line,1205}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]. Last event: {state_timeout,reconnect}. State: {disconnected,{d,'$mria_meta_shard',<0.2729.0>,undefined,undefined,undefined,0,undefined,undefined,false}}.
[error] crasher: initial call: mria_rlog_replica:init/1, pid: <0.2730.0>, registered_name: '$mria_meta_shard', exit: {{timeout,{gen_server,call,[mria_lb,core_nodes,30000]}},[{gen_server,call,3,[{file,"gen_server.erl"},{line,247}]},{mria_rlog_replica,try_connect,2,[{file,"mria_rlog_replica.erl"},{line,378}]},{mria_rlog_replica,handle_reconnect,1,[{file,"mria_rlog_replica.erl"},{line,341}]},{gen_statem,loop_state_callback,11,[{file,"gen_statem.erl"},{line,1205}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}, ancestors: [<0.2729.0>,mria_shards_sup,mria_rlog_sup,mria_sup,<0.1906.0>], message_queue_len: 0, messages: [], links: [<0.2729.0>], dictionary: [{'$logger_metadata$',#{domain => [mria,rlog,replica],shard => '$mria_meta_shard'}}], trap_exit: true, status: running, heap_size: 6772, stack_size: 29, reductions: 51615; neighbours:
[error] Supervisor: {<0.2729.0>,mria_replicant_shard_sup}. Context: child_terminated. Reason: {timeout,{gen_server,call,[mria_lb,core_nodes,30000]}}. Offender: id=replica,pid=<0.2730.0>.
[error] Supervisor: {<0.2729.0>,mria_replicant_shard_sup}. Context: shutdown. Reason: reached_max_restart_intensity. Offender: id=replica,pid=<0.2730.0>.
[error] Supervisor: {local,mria_shards_sup}. Context: child_terminated. Reason: shutdown. Offender: id='$mria_meta_shard',pid=<0.2729.0>.
[info] msg: starting_rlog_shard, mfa: mria_rlog_replica:init/1, line: 93, shard: '$mria_meta_shard'
[info] msg: state_change, mfa: mria_rlog_replica:handle_state_trans/3, line: 439, from: disconnected, shard: '$mria_meta_shard', to: disconnected
[warning] msg: Dashboard monitor error, mfa: emqx_dashboard_monitor:current_rate/1, line: 144, reason: {noproc,{gen_server,call,[emqx_dashboard_monitor,current_rate,5000]}}
```

### What did you expect to happen?

The node would start and rejoin the cluster

### EMQX version

<details>

```console
$ ./bin/emqx_ctl broker
sysdescr  : EMQX
version   : 5.0.24
```

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.6 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.6 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
$ uname -a
Linux 5.15.0-1041-azure #48~20.04.1-Ubuntu SMP Wed Jun 21 15:03:04 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Log files

<details>

</details>

**Comment by qzhuyan** [comment]
could you get us more logs? need some context.

**Comment by ieQu1** [comment]
> ` [error] Supervisor: {<0.2729.0>,mria_replicant_shard_sup}. Context: child_terminated. Reason: {timeout,{gen_server,call,[mria_lb,core_nodes,30000]}}. Offender: id=replica,pid=<0.2730.0>.`

Duplicate of https://github.com/emqx/emqx/issues/11071

**Comment by jkominiakgps** [comment]
@qzhuyan  - the logs I included above are the only logs being generated for the node trying to join the cluster, they simply repeat.  Are there other logs you'd like me to provide or any other detail/context I can share?  Seeing the 1107 bug, it seems very similar.  We established the cluster awhile back and I had a core node leave and join the cluster with out issue.  Since then we've added some rules, flows, and a data bridge and now it appears I'm unable to get any replicant nodes to join the cluster.  

Is there any kind of work around or fix for this currently?
